libs/util/git/src/providers/aws/aws-code-commit.service.ts

```diff
@@ -60,81 +68,185 @@ export class AwsCodeCommitService implements GitProvider {
     });
   }
 
+  private isRequiredValid<T>(
+    value: T | null | undefined
+  ): value is Required<T> {
+    if (value === null || value === undefined) {
+      return false;
+    }
+
+    const props = Object.getOwnPropertyNames(value);
+
+    if (props.length > 0) {
+      return props.every((prop) => {
+        const propValue = value[prop];
+        return propValue !== null && propValue !== undefined;
+      });
+    }
+    return false;
+  }
+
   async init(): Promise<void> {
-    throw new Error("Method not implemented.");
+    throw NotImplementedError;
   }
   async getGitInstallationUrl(amplicationWorkspaceId: string): Promise<string> {
-    throw new Error("Method not implemented.");
+    throw NotImplementedError;
   }
   async getCurrentOAuthUser(accessToken: string): Promise<CurrentUser> {
-    throw new Error("Method not implemented.");
+    throw NotImplementedError;
   }
   async getOAuthTokens(authorizationCode: string): Promise<OAuthTokens> {
-    throw new Error("Method not implemented.");
+    throw NotImplementedError;
   }
   async refreshAccessToken(): Promise<OAuthTokens> {
-    throw new Error("Method not implemented.");
+    throw NotImplementedError;
   }
   async getGitGroups(): Promise<PaginatedGitGroup> {
-    throw new Error("Method not implemented.");
+    throw NotImplementedError;
   }
+
   async getRepository(
     getRepositoryArgs: GetRepositoryArgs
   ): Promise<RemoteGitRepository> {
-    throw new Error("Method not implemented.");
+    const { repositoryName } = getRepositoryArgs;
+    const command = new GetRepositoryCommand({
+      repositoryName,
+    });
+    const { repositoryMetadata } = await this.awsClient.send(command);
+
+    if (this.isRequiredValid(repositoryMetadata)) {
+      return {
+        admin: false,
+        defaultBranch: repositoryMetadata.defaultBranch,
+        fullName: repositoryMetadata.repositoryName,
+        name: repositoryMetadata.repositoryName,
+        private: true,
+        url: repositoryMetadata.cloneUrlHttp,
+        groupName: null,
```

Comment:
Do you want to type this returned object explicitly as RemoteGitRepository?

Reply:
Not needed since we defined the return type for the function and the compiler will complain if this method returns a different object